provisioner/shell/provisioner.go

```diff
@@ -345,19 +344,22 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 			return err
 		}
 
-		if !p.config.SkipClean {
+		if p.config.SkipClean {
+			continue
+		}
 
-			// Delete the temporary file we created. We retry this a few times
-			// since if the above rebooted we have to wait until the reboot
-			// completes.
-			err = p.cleanupRemoteFile(p.config.RemotePath, comm)
-			if err != nil {
-				return err
-			}
-			err = p.cleanupRemoteFile(p.config.envVarFile, comm)
-			if err != nil {
-				return err
-			}
+		// Delete the temporary file we created. We retry this a few times
+		// since if the above rebooted we have to wait until the reboot
+		// completes.
+		if err := p.cleanupRemoteFile(p.config.RemotePath, comm); err != nil {
+			return err
+		}
+
+	}
+
+	if !p.config.SkipClean {
```

Comment:
Looks like we need to keep this file around if SkipClean is true so wrapping the deletion here. Something doesn't feel right code wise about the nested if here, but I might just be nitpicking. Open to any suggestions on how to clean this up if needed :smile:

Reply:
Feels a bit weird but I don't see a way that is obviously more elegant. I guess you could defer the call to delete from inside the loop, but then we wouldn't retrieve the err (not a dealbreaker since it should probably not error the whole provision run anyway), and we'd have to check and only run the defer on the first loop or something to prevent a bunch of defers piling up.

Reply:
I really do like that option. But our future selves will probably be like why didn't you just use an if :laughing: Thanks for the review!

Reply:
yeah, and it's the same amount of code complexity either way. Just leave it as it is here.